src/native_background.ts

```diff
@@ -490,6 +492,9 @@ export async function getConfElsePrefElseDefault(
     return option
 }
 
+// FIXME: The "writePref" message below currently will lead to
```

Comment:
I can't reproduce this bug and I don't see how this could happen when looking at the code. The function gets the whole file, replaces the setting that needs to be replaced or appends a new setting and then writes the whole file again.
I'm very surprised that this happened to you, how can I reproduce it?

Reply:
https://github.com/cmcaine/tridactyl/blob/335e994349da29dc56e71565f981055f2112bb6f/native/native_main.py#L452-L454

The overwriting happens in the snippet above. The steps to reproduce this would be:

1. Drop your own custom `user.js` in your Firefox profile folder
2. From Tridactyl, run `:fixamo`

Thanks.

Reply:
I can't reproduce this bug on the master branch :/
When my `user.js` looks like this:
```js
// Hello world!

user_pref("general.Smoothscroll", false);
```
And I run `:fixamo`, `user.js` becomes this:
```js
// Hello world!

user_pref("general.Smoothscroll", false);

user_pref("privacy.resistFingerprinting.block_mozAddonManager", true);

user_pref("extensions.webextensions.restrictedDomains", "");
```
Are you running Windows? Perhaps this is caused by cache consistency issues in NTFS?

Reply:
@glacambre 

> I can't reproduce this bug on the master branch :/

I think your observation is _correct_. 

The `write()` in Python is definitely destructive, but there are additional checks in `writePref()` in `native_background.ts` that cross-checks what to append/overwrite that I overlooked.

Thanks for double-checking.

https://github.com/cmcaine/tridactyl/blob/335e994349da29dc56e71565f981055f2112bb6f/native/native_main.py#L452-L454

https://github.com/cmcaine/tridactyl/blob/335e994349da29dc56e71565f981055f2112bb6f/src/native_background.ts#L501-L510